src/env_universal_common.rs

```diff
@@ -494,45 +493,43 @@ impl EnvUniversal {
         res_fd
     }
 
-    fn open_temporary_file(&mut self, directory: &wstr, out_path: &mut WString) -> AutoCloseFd {
+    fn open_temporary_file(&mut self, directory: &wstr) -> nix::Result<(OwnedFd, WString)> {
         // Create and open a temporary file for writing within the given directory. Try to create a
         // temporary file, up to 10 times. We don't use mkstemps because we want to open it CLO_EXEC.
         // This should almost always succeed on the first try.
         assert!(!string_suffixes_string(L!("/"), directory));
 
-        let mut saved_errno = Errno(0);
         let tmp_name_template = directory.to_owned() + L!("/fishd.tmp.XXXXXX");
-        let mut result = AutoCloseFd::empty();
-        let mut narrow_str = CString::default();
+        let mut result: nix::Result<(OwnedFd, CString)> = Err(nix::Error::UnknownErrno);
         for _attempt in 0..10 {
-            if result.is_valid() {
+            result = fish_mkstemp_cloexec(wcs2zstring(&tmp_name_template));
+            if result.is_ok() {
                 break;
             }
-            let (fd, tmp_name) = fish_mkstemp_cloexec(wcs2zstring(&tmp_name_template));
-            result.reset(fd);
-            narrow_str = tmp_name;
-            saved_errno = errno();
         }
-        *out_path = str2wcstring(narrow_str.as_bytes());
 
-        if !result.is_valid() {
+        if let Err(errno) = result {
             FLOG!(
                 error,
                 wgettext_fmt!(
                     "Unable to open temporary file '%ls': %s",
-                    out_path,
-                    saved_errno.to_string()
+                    tmp_name_template,
```

Comment:
This is a change in behavior, reading man pages for `mkstemp` /`mkostemp` suggests that after error occurs, the template string can be left in undefined state. So I figure we better not rely on that.

I'm pretty sure it puts the file path in there, but that's not defined as far as I can tell.